pkg/kv/dist_sender.go

```diff
@@ -59,6 +60,47 @@ const (
 	defaultSenderConcurrency = 500
 )
 
+var (
+	metaDistSenderSentCount = metric.Metadata{
+		Name: "distsender.rpc.sent",
+		Help: "Total Number of RPCs sent of distsender",
+	}
+	metaDistSenderLocalSentCount = metric.Metadata{
+		Name: "distsender.rpc.sent-local",
+		Help: "Total Number of local RPCs sent of distsender",
+	}
+	metaDistSenderPartialBatchCount = metric.Metadata{
+		Name: "distsender.rpc.partial-batch",
+		Help: "Total Number of partial Batchs of distsender",
+	}
+	metaDistSenderNotLeaseHolderErrCount = metric.Metadata{
+		Name: "distsender.rpc.errors.notleaseholder",
+		Help: "Total Number of NotLeaseHolderErrors within distsender",
+	}
+)
+
+// DistSenderMetrics is the set of metrics for a given distributed sender.
+type DistSenderMetrics struct {
+	registry               *metric.Registry
+	SentCount              *metric.Counter
+	LocalSentCount         *metric.Counter
+	PartialBatchCount      *metric.Counter
+	NotLeaseHolderErrCount *metric.Counter
+}
+
+func newDistSenderMetrics() *DistSenderMetrics {
+	registry := metric.NewRegistry()
```

Comment:
this doesn't do anything. You're creating a new registry and adding metrics to it, but you're not telling anyone about the registry.
You can do one of the following:
* pass an existing registry when creating the DistSender
* after creating the DistSender, call `AddMetricStruct(distsender.Metrics())`

the second one might be easiest, see `pkg/server/server.go` for how we do it with other objects.

You also need to make sure that your metrics exist by starting a node and checking `/_status/vars`. Debug information about metrics found can be seen by running `cockroach start --vmodule=registery=2`